src/main/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChange.java

```diff
@@ -246,131 +278,242 @@ public String toJson()
-            }
+                if (updatedEdgeView.start() != null)
+                {
+                    ((CompleteEdge) beforeViewUpdatesOnly)
+                            .withStartNodeIdentifier(beforeEdgeView.start().getIdentifier());
+                }
+                if (updatedEdgeView.end() != null)
+                {
+                    ((CompleteEdge) beforeViewUpdatesOnly)
+                            .withEndNodeIdentifier(beforeEdgeView.end().getIdentifier());
+                }
+                break;
+            /*
+             * Line specific updates. The only Line-specific field is the polyline.
+             */
+            case LINE:
+                final Line updatedLineView = (Line) this.updatedView;
+                final Line beforeLineView = (Line) beforeViewFromAtlas;
+                beforeViewUpdatesOnly = CompleteLine.shallowFrom(updatedLineView);
+                if (updatedLineView.asPolyLine() != null)
+                {
+                    ((CompleteLine) beforeViewUpdatesOnly)
+                            .withPolyLine(beforeLineView.asPolyLine());
+                }
+                break;
+            /*
+             * Node specific updates. The Node-specific fields are the location and the in/out edge
+             * sets.
+             */
+            case NODE:
+                final Node updatedNodeView = (Node) this.updatedView;
+                final Node beforeNodeView = (Node) beforeViewFromAtlas;
+                beforeViewUpdatesOnly = CompleteNode.shallowFrom(updatedNodeView);
+                if (updatedNodeView.getLocation() != null)
+                {
+                    ((CompleteNode) beforeViewUpdatesOnly)
+                            .withLocation(beforeNodeView.getLocation());
+                }
+                if (updatedNodeView.inEdges() != null)
+                {
+                    ((CompleteNode) beforeViewUpdatesOnly).withInEdges(beforeNodeView.inEdges());
+                }
+                if (updatedNodeView.outEdges() != null)
+                {
+                    ((CompleteNode) beforeViewUpdatesOnly).withOutEdges(beforeNodeView.outEdges());
+                }
+                break;
+            /*
+             * Point specific updates. The only Point-specific field is the location.
+             */
+            case POINT:
+                final Point updatedPointView = (Point) this.updatedView;
+                final Point beforePointView = (Point) beforeViewFromAtlas;
+                beforeViewUpdatesOnly = CompletePoint.shallowFrom(updatedPointView);
+                if (updatedPointView.getLocation() != null)
+                {
+                    ((CompletePoint) beforeViewUpdatesOnly)
+                            .withLocation(beforePointView.getLocation());
+                }
+                break;
+            /*
+             * Relation specific updates. The only Relation-specific field is the member list.
+             */
+            /*
```

Comment:
What to do about this allKnownOsmMembers case? I guess I just don't understand this aspect of relations well at all. (What is it, and why do we need it?)